test/Unipool.js

```diff
@@ -217,11 +224,146 @@ contract('Unipool', function ([_, wallet1, wallet2, wallet3, wallet4]) {
             expect(await this.pool.earned(wallet1)).to.be.bignumber.equal('0');
             expect(await this.pool.earned(wallet2)).to.be.bignumber.equal('0');
 
-            await timeIncreaseTo(this.started.add(time.duration.weeks(1)));
+            await timeIncreaseTo(this.started.add(time.duration.days(30)));
 
             expect(await this.pool.rewardPerToken()).to.be.bignumber.almostEqualDiv1e18(web3.utils.toWei('2500'));
             expect(await this.pool.earned(wallet1)).to.be.bignumber.almostEqualDiv1e18(web3.utils.toWei('2500'));
             expect(await this.pool.earned(wallet2)).to.be.bignumber.almostEqualDiv1e18(web3.utils.toWei('7500'));
         });
+
+        // TODO: if for some period the total supply is zero, rewards for that period are lost!
```

Comment:
I think this is fair, let's add a test for this case and remove the TODO

Reply:
Is it? Unless I’m missing something, the tokens would be lost (stuck in the contract). The tests below are already for that case, see that the reward contributed is 50k while the reward earned is 25k.

Reply:
Oh, I thought these were simply carried for the next period, I see what you mean now. No, definitely it is not ideal. 
I haven't looked that much into it, would it be too hard to simply make those rewards available for the next period?

Reply:
It was discussed offline, we’ll leave it as is for now. There’s a simple workaround: we can make sure ourselves that it’s not zero by staking a tiny amount of tokens.